src/traits/default_impls.rs

```diff
@@ -261,36 +262,49 @@ impl DefaultCellCollector {
             indexer_client,
             ckb_client,
             offchain: OffchainCellCollector::default(),
+            acceptable_indexer_leftbehind: 1,
         }
     }
 
-    /// Check if ckb-indexer synced with ckb node. This will check every 50ms for 10 times (500ms in total).
+    /// THe acceptable ckb-indexer leftbehind block number (default = 1)
+    pub fn acceptable_indexer_leftbehind(&self) -> u64 {
+        self.acceptable_indexer_leftbehind
+    }
+    /// Set the acceptable ckb-indexer leftbehind block number
+    pub fn set_acceptable_indexer_leftbehind(&mut self, value: u64) {
+        self.acceptable_indexer_leftbehind = value;
+    }
+
+    /// Check if ckb-indexer synced with ckb node. This will check every 50ms for 100 times (more than 5s in total, since ckb-indexer's poll interval is 2.0s).
     pub fn check_ckb_chain(&mut self) -> Result<(), CellCollectorError> {
-        let tip_header = self
+        let mut tip_number = self
             .ckb_client
-            .get_tip_header()
+            .get_tip_block_number()
             .map_err(|err| CellCollectorError::Internal(err.into()))?;
-        let tip_hash = tip_header.hash;
-        let tip_number = tip_header.inner.number;
-        let mut retry = 10;
+
+        let mut retry = 100;
```

Comment:
I think we can skip ckb node tip fetching in the retry loop and simplify the code to:  

```suggestion
        for _retry in 0..100 {
            match self
                .indexer_client
                .get_tip()
                .map_err(|err| CellCollectorError::Internal(err.into()))?
            {
                Some(Tip { block_number, .. }) => {
                    if tip_number.value()
                        > block_number.value() + self.acceptable_indexer_leftbehind
                    {
                        thread::sleep(Duration::from_millis(50));
                    } else {
                        return Ok(());
                    }
                }
                None => {
                    return Err(CellCollectorError::Other(
                        "ckb-indexer server not synced".to_owned().into(),
                    ));
                }
            }
        }
```